Added support for the [TTT2] Statistics-Addon

Henotu opened the pull request:
ttt2-last-man-standing tacks and displays a few stats with the help of the "[TTT2] Statistics-Addon"
I added a completely new file to handle a GUI-Integration like it is described [here](https://github.com/Henotu/TTT2-Statistics/wiki/Add-your-own-Addon!).

Also, somehow some lines got changed by a space even though I didn't caused that.

If the statistics-addon isn't installed nothing will be tracked or break (tested it).

This is what LMS will look like in the GUI:
![LMS](https://user-images.githubusercontent.com/34514481/80613054-0b925900-8a3d-11ea-9408-dffd94998ea6.jpg)

Commit:
Added support for the [TTT2] Statistics-Addon

menof36go commented on lua/autorun/sh_lms_statistics.lua:
```diff
@@ -0,0 +1,44 @@
+  end)
+
+  function LastInnoStanding()
+    LastInno = true
```
What's the purpose of a function like this? Seems rather pointless

menof36go commented on lua/autorun/sh_lastManStanding_init.lua:
```diff
@@ -42,6 +42,8 @@ if SERVER then
         end
     end)
     util.AddNetworkString("ttt_lms_notify")
+    util.AddNetworkString("lms_statistics_inno")
+    util.AddNetworkString("lms_statistics_reveal")
```
Please refactor to ttt_lms_... to be consistent with ttt_lms_notify

menof36go commented on lua/autorun/sh_lastManStanding_init.lua:
```diff
@@ -126,6 +128,9 @@ if SERVER then
             net.Start("ttt_lms_notify")
             net.WriteBool(true)
             net.Broadcast()
+            net.Start("lms_statistics_reveal")
+            net.WriteEntity(caller)
+            net.Broadcast()
```
Does this need to be broadcast? Why not go with net.Send(caller) instead

Henotu replied:
Thanks for the suggestion, I actually haven't thought of that

menof36go commented on lua/autorun/sh_lms_statistics.lua:
```diff
@@ -0,0 +1,44 @@
+
+  hook.Add("TTT2FinishedLoading", "ttt_lms_statistics", function()
+    if isfunction(AddYourStatisticsAddon) then
+      AddYourStatisticsAddon("Last Man Standing", LMSStatisticsIntegration)
```
Unrelated, but "AddYourStatisticsAddon" is a really weird name for any function

Henotu replied:
I know what you mean, but I couldn't think of a better name

Commit:
Changed the global function LastInnoStanding() to
the global variable LastInnoStanding

Fixed NetworkString names

Changed net.Broadcast() into net.Send() in some cases

menof36go commented on lua/autorun/sh_lastManStanding_init.lua:
```diff
@@ -128,9 +128,9 @@ if SERVER then
-            net.WriteEntity(caller)
-            net.Broadcast()
+            net.Start("ttt_lms_reveal")
+            net.WriteBool(true)
```
I think net.WriteBool can be removed

menof36go commented on lua/autorun/sh_lastManStanding_init.lua:
```diff
@@ -201,9 +201,9 @@ if SERVER then
-            net.WriteEntity(caller)
-            net.Broadcast()
+            net.Start("ttt_lms_innocent")
+            net.WriteBool(true)
```
I think net.WriteBool can be removed

menof36go commented on lua/autorun/sh_lastManStanding_init.lua:
```diff
@@ -269,16 +269,16 @@ if CLIENT then
-    net.Receive("lms_statistics_reveal",function()
-      if (net.ReadEntity() == LocalPlayer()) and (isfunction(StatisticsUpdatePData)) then
+    net.Receive("ttt_lms_reveal",function()
+      if (net.ReadBool()) and (isfunction(StatisticsUpdatePData)) then
```
I think net.ReadBool() can be removed. If you are not the player the ttt_lms_reveal net call is intended for, you will never receive this anyways.

menof36go commented on lua/autorun/sh_lastManStanding_init.lua:
```diff
@@ -269,16 +269,16 @@ if CLIENT then
-    net.Receive("lms_statistics_inno",function()
-      if (net.ReadEntity() == LocalPlayer()) and (isfunction(StatisticsUpdatePData)) then
+    net.Receive("ttt_lms_innocent",function()
+      if (net.ReadBool()) and (isfunction(StatisticsUpdatePData)) then
```
I think net.ReadBool() can be removed. If you are not the player the ttt_lms_reveal net call is intended for, you will never receive this anyways.

menof36go commented:
Good on you for refactoring the broadcast to send. Always keep in mind that some players might have slow internet connections (looking at no one specifically except Ohrhusten) so we want to send only what is necessary

Commit:
Removed unnecessary net.WriteBool() and net.ReadBool()

ys3d commented:
Maybe you should remove some of the random whitespaces

Commit:
Added support for an upcoming update of the statistics-addon
